File: lesson-09-spring-rest1/src/main/java/ru/aberezhnoy/service/CartService.java
package ru.aberezhnoy.service;

import org.springframework.stereotype.Service;
import ru.aberezhnoy.persist.Product;
import ru.aberezhnoy.service.dto.LineItem;
import ru.aberezhnoy.service.dto.ProductDto;

import java.util.List;

@Service
public interface CartService {
    List<LineItem> showAll();

    void addToCart(ProductDto productDto);

    void removeFromCart(LineItem lineItem);
}

File: frontend/src/app/services/cart.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {LineItem} from "../model/lineItem";
import {Product} from "../model/product";

@Injectable({
  providedIn: 'root'
})
export class CartService {

  constructor(private http: HttpClient) {
  }

  public showAll() {
    return this.http.get<LineItem[]>('api/v1/cart');
  }

  public addToCart(product: Product) {
    return this.http.post<Product>('api/v1/cart', product);
  }

  public removeFromCart(product: Product) {
    return this.http.delete<Product>('api/v1/cart', ({
      body: product
    }));
  }
  //
  // public delete(product: Product) {
  //   return this.http.delete<Product>('api/v1/cart');
  // }
}
